KalturaWowzaServer/src/main/java/com/kaltura/media_server/services/KalturaStreamsDataPersistence.java

```diff
@@ -0,0 +1,131 @@
+import com.wowza.wms.stream.*;
+import org.apache.log4j.Logger;
+
+import java.util.Objects;
+import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.List;
+
+import org.apache.commons.lang3.tuple.*;
+
+
+/**
+ * Created by lilach.maliniak on 29/01/2017.
+ */
+public class KalturaStreamsDataPersistence {
+
+	private static Logger logger = Logger.getLogger(KalturaStreamsDataPersistence.class);
+	private static WeakHashMap<KalturaEntryIdKey, ConcurrentHashMap<String, Object>> entryIdToKalturaLiveEntryMap = new WeakHashMap<>();
+
+
+	public static Pair<Object, KalturaEntryIdKey> getEntry(IClient client, String subKey) throws Exception{
+
+		WMSProperties properties = client.getProperties();
+		KalturaEntryIdKey entryIdKey = null;
+		List<IMediaStream> names = client.getPlayStreams();
+		String streamName = names.size() > 0 ? names.get(0).getName() : "client with no stream";
+
+		Object value = getEntry(subKey, streamName);
+
+		return Pair.of(value, entryIdKey);
+
+	}
+
+	public static Pair<Object, KalturaEntryIdKey> getEntry(String subKey, String streamName, Object defaultValue) {
+
+		try {
+			return getEntry(subKey, streamName, defaultValue);
+		} catch (Exception e) {
+			logger.error(e + " returning the default value for sub key " + subKey);
+			String entryId = Utils.getEntryIdFromStreamName(streamName);
+			KalturaEntryIdKey entryIdKey = new KalturaEntryIdKey(entryId);
+
+			return Pair.of(defaultValue, entryIdKey);
+		}
+	}
+
+	public static Pair<Object, KalturaEntryIdKey> getEntry(String subKey, String streamName) throws Exception {
+
+		String entryId = Utils.getEntryIdFromStreamName(streamName);
+		KalturaEntryIdKey entryIdKey = new KalturaEntryIdKey(entryId);
+
+		Object value = getEntry(entryIdKey, subKey, streamName);
+		Pair<Object, KalturaEntryIdKey> result = Pair.of(value, entryIdKey);
+		return result;
+	}
+
+	public static Object getEntry(KalturaEntryIdKey entryIdKey, String subKey, String uniqueName) throws Exception {
+
+		Object value = null;
+
+		synchronized (entryIdToKalturaLiveEntryMap) {
+			ConcurrentHashMap<String, Object> entryMap = entryIdToKalturaLiveEntryMap.get(entryIdKey);
+
+			if (entryMap == null) {
+				throw new Exception("(" + uniqueName + ") ("+ entryIdKey.getEntryId() +") failed to get value for " + subKey + ". The key (" + entryIdKey.getEntryId() + ") doesn't exist in map.");
+			} else if (entryMap.get(subKey) == null) {
+				throw new Exception("(" + uniqueName + ") ("+ entryIdKey.getEntryId() +") failed to get value for " + subKey + ". The sub key (" + subKey + ") doesn't exist in map.");
+			}
+
+			value = entryMap.get(subKey);
+		}
+
+		logger.debug("(" + uniqueName + ") ("+ entryIdKey.getEntryId() +") successfully got entry");
+
+		return value;
+	}
+
+	// add entry on connect (after successful authentication)
+	public static Pair<Object, KalturaEntryIdKey> addEntry(String entryId, String subKey, Object value, String partnerId) throws Exception {
```

Comment:
rename to SetPropoery, and I'd expect it to return Object and not the actual KalturaEntryIdKey